New tutorial for SAP Fiori elements Mockserver and OPA test library

ARiesterer opened the pull request:
Automatic commit: Move 'fiori-tools-mockserver-opa', 'fiori-tools-mockserver-opa-testing' from QA to Production

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa/fiori-tools-mockserver-opa.md:
```diff
@@ -0,0 +1,217 @@
+description: Using mockserver and advance features in running SAP Fiori Elements application.
+auto_validation: true
+time: 30
+tags: [ tutorial>beginner, software-product>sap-fiori-tools, products>sap-fiori-elements, products>sap-business-application-studio ]
```
update tags. products -> software-product

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa/fiori-tools-mockserver-opa.md:
```diff
@@ -0,0 +1,217 @@
+primary_tag: software-product>sap-fiori
+---
+
+## Prerequisites
```
Add periods at the end of each line

MichaelCzcz reviewed (changes requested):
Thank you for submitting this PR.
There are some style issues that you should fix before I can merge it.
For some I have commented on the respective line.

One thing that sticks out most are those huge screenshots that you are using. According to the style guide images should be not bigger than 1200x500px.
They should also only show relevant information. Your browser's address bar, your bookmarks bar etc. are thing that you should not include in screenshots unless it is absolutely necessary.
You also don't have to take a screenshot of the full window. Just pick the parts that are relevant. You can use Snipping Tool or Snagit to create proper screenshots.

D051664 commented:
Hi Michael,
the requested changes are committed.
Please see the [commit](https://github.com/sap-tutorials/Tutorials/pull/7908/commits/d1ae4a614c1840c18271cff358c04b818a38f643) and trigger the release of the tutorial.
Thanks and best regards
Andreas

ARiesterer commented:
author information updated

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+
+Steps 5 and 6 of this tutorial demonstrate the adoption of the SAP Fiori elements for OData V4 OPA test library.  
+
+The OPA test journey available in file `OpaJourney.js` is just a starting point for you to enhance the tests for the list report and the object page. This journey currently starts the application after the test data is restored. Then, the list report table is loaded and a function checks whether data available in the table. As a last step, the application is closed.
```
"list report table" - Please use hyphens in compound words to improve readability. Otherwise it could mean that you want the user to "use list" and "report table". Please also change all other ocurrences.

ARiesterer replied:
done

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+
+3. Execute an action on a selected item in the list report
+
+    Before executing standard or custom bound actions, you can select an item in the list report table by using the test function `iSelectRows()`. As with the check function for rows, you can determine the selection by a condition. The action to be executed will be the deletion of a travel item. Do this by calling the test function `iExecuteDelete()`.
```
"custom bound actions" - This one may also need hyphens.

ARiesterer replied:
done

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
````diff
@@ -0,0 +1,606 @@
+    ```
+    Since these actions are related to the list report table, you use the identifier `onTable()` to access the corresponding functions. You don´t need to provide a parameter for `onTable()`, since there´s only one table available on the list report. For more functions related to the list report table, please have a look at the content of **TableActions** and **TableAssertions** within the API doc ...
+
+    >Note: There are several test functions for pressing and checking standard buttons like Edit, Delete and Save, that are managed by the SAP Fiori elements framework. Such test functions are e.g. iExecuteEdit/iCheckEdit, iExecuteDelete()/iCheckDelete(), etc. The test library provides the functions iExecuteAction() and iCheckAction() to perform action and assertions on custom action buttons. To  ...
````
Don't start a note with "Note:" - You are already using the proper formatting that shows the reader that he is reading a note.
Please check all other notes and change them accordingly.

ARiesterer replied:
done

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+    >Note: There are several test functions for pressing and checking standard buttons like Edit, Delete and Save, that are managed by the SAP Fiori elements framework. Such test functions are e.g. iExecuteEdit/iCheckEdit, iExecuteDelete()/iCheckDelete(), etc. The test library provides the functions iExecuteAction() and iCheckAction() to perform action and assertions on custom action buttons. To  ...
+
+
+    Once the journey reaches the delete function it stops and after some time you´ll see an error message in the log area (see screenshot below). The standard delete action shows a dialog which asks the user to confirm the deletion. This dialog also needs to be handled by the OPA test, i.e. you have to confirm the deletion by pressing the **Delete** button.
```
missing comma after "function"
missing comma before "which" - Please check all other "whiches" and add missing commas.

ARiesterer replied:
done

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+
+2. Check a field in a form of a section
+
+    You can use the form assertions to check fields within the sections of an object page. As implemented for most of the test functions, you can use both strings and objects as parameters for identifying the UI controls you want to check. When you provide a string, the test functions expect the label of a control to be passed as a value. If an object is provided as the parameter, the internal na ...
```
"You can use the form assertions to check fields " - If the word "from" is needed here, please put it in back-tics.

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+
+    The following example shows both options of passing the parameters to identify the field **Agency** and to check its content. In the first call of the test function `iCheckField()`, strings are used to identify the field. The second call uses the internal names for the section, form and field and also passes the content with the elements `value` and `description`.
+
+    You can use function chaining to enhance the existing checks for the page and the header controls. Using `.and.then.` returns the control to the page to then use the `onForm()` command to check the field.
```
function chaining - use hyphen

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa/fiori-tools-mockserver-opa.md:
```diff
@@ -0,0 +1,222 @@
+
+
+## Details
+### You will learn
```
Please move this section above "Prerequisites"

MichaelCzcz replied:
Please disregard this one. Everything is in its correct place.

ARiesterer replied:
done

ARiesterer replied:
ok - I will take it back as it was

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+author_profile: https://github.com/ARiesterer
+---
+
+## Prerequisites
```
Please move this section below "You will learn"

MichaelCzcz replied:
Please disregard this one. Everything is in its correct place.

ARiesterer replied:
done

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
````diff
@@ -0,0 +1,606 @@
+      ...
+    }
+    ```
+    If this entry is available, you can start the tests using
````
Please do not use notes within a sentence. Please rephrase.

ARiesterer replied:
done

MichaelCzcz commented on tutorials/fiori-tools-mockserver-opa/fiori-tools-mockserver-opa.md:
```diff
@@ -0,0 +1,222 @@
+        "metadataXmlPath": "./AgencyVHMetadata.xml",
+        "mockdataRootPath": "./mockdata",
+        "generateMockData": true
+    }]
```
Is this intended?

ARiesterer replied:
I think, the object shown in the coding fragment is the last one within an array. Since only a part of the array is shown here in the text, it looks as if it is wrong, but in the real config file it is correct.
Therefore I would tend to leave it here as it is.

MichaelCzcz reviewed (changes requested):
Please address the comments and resolve each conversation afterwards.
I did not fully proof-read the texts, so I may not have caught all errors. Please proof-read both tutorials again.

ARiesterer commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+---
+title: Write OPA Tests for an SAP Fiori Elements for OData V4 Application
+description: Reuse test functions from the SAP Fiori elements for OData V4 OPA test library to write OPA tests.
```
fixed in description of the tutorial:
description: Reuse test functions from the SAP Fiori elements for OData V4 OPA test library to write OPA (One Page Acceptance) tests.

ARiesterer commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
```diff
@@ -0,0 +1,606 @@
+
+        This is the preferred option for executing OPA tests. You can store the mock data as json files in the `/localService/mockdata` folder and each time the server and the application are restarted, the same set of data is loaded. This means  the tests are always based on the same set of data.
+
+        For this tutorial, a very simple set of static mock data is used which consists of two json files: `Travel.json` and `Booking.json`. The data that is contained in these files is sufficient to run the test functions used in the tutorial. You can downloaded these files from the GitHub repository with the URL below:
```
done

ARiesterer commented on tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md:
````diff
@@ -0,0 +1,606 @@
+        });
+        ```
+
+        >In line 9 of the coding fragment above, the application is started using the function `iStartMyApp()`. Please add the name (semantic object and action) as a parameter to the function `iStartMyApp("travellist-tile")`. You can find the name of your application in the file **/webapp/test/flpSandbox.html** in the `applications:` parameter, which should look similar to the one listed below. L ...
````
done

ARiesterer commented:
Hi Michael,
I have addressed all issues you listed above and committed.
I already did a proof-read together with our language expert so I guess it should be ok now for my part.
For Reginas part I also did the corrections with the last commit. Since she is an English native speaker, we also should be ok for her part.
